sdk/appcenter-data/src/main/java/com/microsoft/appcenter/data/Data.java

```diff
@@ -1014,6 +1014,10 @@ private TokenResult getCachedToken(String partitionName) {
             completeFuture(getInvalidPartitionDataException(partition), result);
             return true;
         }
+        if (documentId == null || documentId.isEmpty() || documentId.contains("#")) {
```

Comment:
I think space won't work too :) because after urlencode the result is `+` but we want `%20`

Reply:
space works, cosmos db replaces them by `-`.
Only empty string does not work

Reply:
but when I manually change the document id on portal like add an empty space... when I read it, it returns 404...